src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -41,20 +42,56 @@ public function __construct(array $sendersMap, ContainerInterface $sendersLocato
      */
     public function getSenders(Envelope $envelope): iterable
     {
-        $seen = [];
+        $senderAliases = [];
 
         foreach (HandlersLocator::listTypes($envelope) as $type) {
+            $typeSenderAliases = [];
+
             foreach ($this->sendersMap[$type] ?? [] as $senderAlias) {
-                if (!\in_array($senderAlias, $seen, true)) {
-                    if (!$this->sendersLocator->has($senderAlias)) {
-                        throw new RuntimeException(sprintf('Invalid senders configuration: sender "%s" is not in the senders locator.', $senderAlias));
-                    }
-
-                    $seen[] = $senderAlias;
-                    $sender = $this->sendersLocator->get($senderAlias);
-                    yield $senderAlias => $sender;
-                }
+                $typeSenderAliases[] = $senderAlias;
+            }
+
+            if (empty($typeSenderAliases)) {
+                $typeSenderAliases = $this->getSendersFromAttributes($type);
+            }
+
+            $senderAliases = array_merge($senderAliases, $typeSenderAliases);
```

Comment:
If I understand it correctly, IF even one *real* route exists for the message (something in `$this->sendersMap[$type]`), then we do NOT also add the `#[Transport]` senders? So basically any config would *override* the attribute?

I think that is... probably the correct behavior. But what about the `*` catch-all route in `messenger.yaml` (or an interface)? I'm wondering if someone will have the use-case where they generally use the attribute, but then they decide to, in addition to the transport in each class, route all messages (or all messages for an interface) to a 2nd transport via the routing config in `messenger.yaml`. 

So, I'm conflicted on this point.